### ISSUE TITLE
How to ZIP the Electron *Mac* files like Electron does it? (From Windows.)

### ISSUE THREAD

**Issue by johnjohn1231** [electron/electron#11634]
(Since my last issue was "closed" without my question being answered, I'll try one more time before I give up on trying to make a Mac version of my application.)

When I use 7-Zip on Windows to extract the Mac Electron ZIP archive, add my files and then ZIP it back up, it loses some symlinks or something. There's some kind of difference between Windows and Mac that makes the resulting ZIP file not quite right. Specifically, if you download the following ZIP file straight to your Mac and extract it, the resulting program WILL run fine: https://github.com/electron/electron/releases/download/v1.7.10/electron-v1.7.10-darwin-x64.zip

... BUT if you download the same file to Windows, extract it with 7-Zip, then ZIP it back up and send it to the Mac, the resulting program will FAIL to run.

With the Linux version, I had a similar issue, which was solved by making Linux's package a .tar instead of .zip. The same "trick" didn't work on Mac, however.

I've asked *many* people and searched endlessly without even finding this mentioned anywhere. I cannot understand why this happens or how to solve it. (And yes, I must use the Windows machine and that alone.)

I've also vacuumed the 7-Zip manual without finding any mention of this. Somebody did mention that Macs and Windows interpret UTF-8 slightly differently or something, but I'm 99% sure it's some weird "symbolic links" that don't get packed/unpacked properly. Can the Electron developers maybe start making the Mac version without such error-prone symlinks?

Basically, how are the Electron developers making the working Mac ZIP archive? Do they also use 7-Zip? Something else? Note once again that it has to work from Windows, and automated/scripted. Thanks. I hope this time somebody understands the issue!

**Comment by MarshallOfSound** [comment]
Your previous issue was closed and answered https://github.com/electron/electron/issues/11622

It should also be pointed out that issues in this repository are not the place for questions, you should be asking those on our community services such as the forum and slack channel.

As @felixrieseberg mentioned in your other issue we recommend you build for a given platform on the target platform.  This is the best way to build for macOS.  This issue at play here is that windows doesn't support symbolic links so by unzipping and rezipping on windows, that Mac specific information is lost.

TLDR. Windows and symlinks don't play nice, build for Mac on Mac to avoid symlinks issues

**Comment by johnjohn1231** [comment]
Open issue.

I said that I'm doing this on WINDOWS. Why not LISTEN?

**Comment by MarshallOfSound** [comment]
@johnjohn1231 Sorry if I wasn't clear enough, what you are asking to do is not technically possible.  As soon as you unzipping on windows and lose those symlinks you can't get them back again.

**Comment by brenca** [comment]
I have to agree with @MarshallOfSound, your issue was answered, now 2 times at least. You can't use 7zip to do this on Windows, and AFAIK you can't use anything else either, because these programs on Windows don't keep the symlinks in the zip files. @felixrieseberg even linked the relevant python code, which shows that on Mac, the zips are created with the zip command, with a `-r -y` flag. According to google, the `-y` flag is responsible for keeping the symlinks in the zip file. 

If you only need to add to the zip file, you could try adding to it without unzipping everything first, that could circumvent this limitation, but I can't guarantee it. If that doesn't work, you are going to need a Mac to build your application. 

(BTW people usually use something like electron-builder to build their apps, if I understand correctly you are manually copying files into the zip. Note that using electron-builder still won't solve your problem, it's just a better way than manual copies.)
